tests/units/units.cpp

```diff
@@ -2004,3 +2145,111 @@ TEST(Units, scalingFactorAcrossStandardUnits)
     EXPECT_EQ(0.001, libcellml::Units::scalingFactor(u1, u2));
     EXPECT_EQ(1000.0, libcellml::Units::scalingFactor(u2, u1));
 }
+
+TEST(Units, scalingFactorChildUnitsNotFound)
+{
+    auto model = libcellml::Model::create("nurseryrhymes");
+    auto u1 = libcellml::Units::create("bo_peep");
+    u1->addUnit("sheep");
+
+    auto u2 = libcellml::Units::create("flock");
+
+    model->addUnits(u1);
+    model->addUnits(u2);
+
+    EXPECT_EQ(0.0, libcellml::Units::scalingFactor(u1, u2));
+}
+
+TEST(Units, scalingFactorGrandchildUnitsNotFound)
+{
+    auto model = libcellml::Model::create("nurseryrhymes");
+    auto u0 = libcellml::Units::create("sheep");
+    u0->addUnit("wool");
+
+    auto u1 = libcellml::Units::create("bo_peep");
+    u1->addUnit("sheep");
+
+    auto u2 = libcellml::Units::create("flock");
+
+    model->addUnits(u0);
```

Comment:
`u0` is not at all used in this test, so why have it?

Reply:
Updated to better show behaviour.

Reply:
Still not clear where `u0` is being used in this test...

Reply:
See full context below.  `u0` is needed to provide a lower generation against which the parent units need to be resolved.

```
TEST(Units, scalingFactorGrandchildUnitsNotFound)
{
    auto model = libcellml::Model::create("nurseryrhymes");

    auto u0 = libcellml::Units::create("sheep");
    u0->addUnit("wool");

    auto u1 = libcellml::Units::create("bo_peep");
    u1->addUnit("sheep");

    auto u2 = libcellml::Units::create("flock");
    u2->addUnit("sheep");

    model->addUnits(u0);
    model->addUnits(u1);
    model->addUnits(u2);

    // Child units of "wool" are not found within "sheep" units.  Comparison should return 0.0.
    EXPECT_EQ(0.0, libcellml::Units::scalingFactor(u1, u2));
}
```

Reply:
👍